topi/python/topi/transform.py

```diff
@@ -338,3 +338,21 @@ def take(a, indices, axis=None):
     if axis is None:
         return cpp.take(a, indices)
     return cpp.take(a, indices, int(axis))
+
+@tvm.tag_scope(tag=tag.INJECTIVE)
```

Comment:
The fuse pattern is OUT_ELEMWISE_FUSABLE
https://github.com/dmlc/tvm/blob/edd378007ddb1adb5451f2a73f60f6ff538b37a8/nnvm/python/nnvm/top/nn.py#L77

Reply:
Very interesting place, I will try to figure it out. For now I've fixed the annotation as you suggested, Thanks!

Reply:
I feel I really need to understand how does patterns/tags/schedulers work in order to do the right thing. Am I correct saying that `@tvm.tag_scope` does nothing when used with `cpp` functions? I see  `tvm/api.py` alone refers to `TagScope` , but in our case the C++ version of `tvm::compute` is used.

Reply:
AFAIK, in cpp case, the default `tag` arguments are used for tagging, e.g. https://github.com/dmlc/tvm/pull/1744/commits/9784c9e06dfc383cb6e6ca70100c38aac24a0252#diff-6bbbb63b8fa5dc215443d88cfafed27aR650 If I am correct, then we probably should remove all `@tvm.tag_scope` declarations from cpp wrappers declared in `topi.transform.py` .

Reply:
You are right. We should remove all tag scopes for cpp wrappers.